src/group_elements.jl

```diff
@@ -1,223 +1,241 @@
-Return conjugation of `g` by `h`, i.e. `h^-1*g*h`.
+    conj(g::G, h::G) where {G <: GroupElement}
+
+Return conjugation of $g$ by $h$, i.e. $h^{-1} g h$.
 """
-Base.conj(g::GEl, h::GEl) where {GEl<:GroupElement} = conj!(similar(g), g, h)
+Base.conj(g::G, h::G) where {G <: GroupElement} = conj!(similar(g), g, h)
 
 @doc Markdown.doc"""
-    ^(g::GEl, h::GEl) where {GEl<:GroupElement}
-Conjugation action of `h` on `g`. See `conj`.
+    ^(g::G, h::G) where {G <: GroupElement}
+
+Alias for `conj`.
 """
-Base.:(^)(g::GEl, h::GEl) where {GEl<:GroupElement} = conj(g, h)
+Base.:(^)(g::G, h::G) where {G <: GroupElement} = conj(g, h)
 
 @doc Markdown.doc"""
-    comm(g::GEl, h::GEl[, Vararg{GEl}...) where GEl<:GroupElement
-Return the commutator `g^-1*h^-1*g*h` of `g` and `h`.
+    comm(g::G, h::G, k::G...) where {G <: GroupElement}
 
-The `Vararg` version returns the repeated (`foldl`) commutator, i.e.
-`comm(g, h, k) == comm(comm(g, h), k)`.
+Return the commutator $[[g, h], ...]$, where $[g, h] = g^{-1} h^{-1} g h$.
 """
-function comm(g::GEl, h::GEl, tail::GEl...) where {GEl<:GroupElement}
+function comm(g::G, h::G, k::G...) where {G <: GroupElement}
     res = comm!(similar(g), g, h)
-    for k in tail
+    for l in k
         res = comm!(res, res, k)
     end
     return res
 end
 
 Base.literal_pow(::typeof(^), g::GroupElement, ::Val{-1}) = inv(g)
 
-Base.:(/)(g::GEl, h::GEl) where {GEl<:GroupElement} =
-    div_right!(similar(g), g, h)
+@doc Markdown.doc"""
+    /(g::G, h::G) where {G <: GroupElement}
+
+Return $g h^{-1}$.
+"""
+Base.:(/)(g::G, h::G) where {G <: GroupElement} = div_right!(similar(g), g, h)
 
-#### Modification possible for performance reasons
+################################################################################
+# Default implementations that (might) need performance modification
+################################################################################
 
 @doc Markdown.doc"""
     similar(g::GroupElement)
-Return an arbitrary (and possibly uninitialized) group element sharing the parent with `g`.
+
+Return a group element sharing the parent with $g$. Might be arbitrary and
+possibly uninitialized.
 """
 Base.similar(g::GroupElement) = one(g)
 
+@doc Markdown.doc"""
+    isone(g::GroupElement)
+
+Return true if $g$ is the identity element.
+"""
 Base.isone(g::GroupElement) = g == one(g)
 
 @doc Markdown.doc"""
-    isequal(g::GEl, h::GEl) where {GEl<:GroupElement}
-The "best effort" equality for group elements.
+    isequal(g::G, h::G) where {G <: GroupElement}
 
-Depending on the group this might, or might not be the correct equality, which can be obtained using `Base.isequal`. Nonetheless the implication `g == h` → `isequal(g, h)` must be always satisfied, i.e. "best effort" equality might return `false` even when group elements are equal.
+Return the "best effort" equality for group elements. Always returns true if
+$g = h$, but might also return true if $g$ is not equal to $h$.
```

Comment:
Haha, I wrote the implication in docstring in the wrong direction ;D

What we want is `isequal(g,h)` → `g==h`, i.e. identity of words implies that group elements are equal, but the former may return false even if `g` and `h` are equal as group elements!

Reply:
Ahh, I see!